ldaptor/protocols/ldap/distinguishedname.py

```diff
@@ -63,59 +51,47 @@ def unescape(s):
             r = r + s[0]
             s = s[1:]
 
-    if have_bytes:
-        return r.encode('utf-8')
-    else:
-        return r
+    return r
 
 
 def _splitOnNotEscaped(s, separator):
     if not s:
         return []
 
-    have_bytes = False
-    if isinstance(s, bytes):
-        s = s.decode('utf-8')
-        have_bytes  = True
-
-
-    r = ['']
+    r = [u'']
     while s:
         first = s[0:1]
 
-        if first == '\\':
+        if first == u'\\':
             r[-1] = r[-1] + s[:2]
             s = s[2:]
         else:
 
             if first == separator:
-                r.append('')
+                r.append(u'')
                 s = s[1:]
-                while s[0:1] == ' ':
+                while s[0:1] == u' ':
                     s = s[1:]
             else:
                 r[-1] = r[-1] + first
                 s = s[1:]
 
-    if have_bytes:
-        return [x.encode('utf-8') for x in r]
-    else:
-        return r
+    return r
 
 
 class InvalidRelativeDistinguishedName(Exception):
     """Invalid relative distinguished name."""
 
     def __init__(self, rdn):
         Exception.__init__(self)
-        self.rdn = rdn
+        self.rdn = rdn.encode('utf-8') if six.PY2 else rdn
```

Comment:
Can we have this it was passed in Py2.

My fear is that if someone is already using this on Py2 with UTF-8 it will pass already encoded UTF-8 and then the code will break.

So instead of looking for Python version we can look for input type... if is not 

Also, as part of this work it would be nice to update the docstring to document the input types.

What is the type of `rdn` ? bytes or Unicode?

Reply:
> `rdn` is `str` :)

better to have this answered in the docstring :)

Reply:
Oh, made more verbose docstring for this exception class :)